cmd/kube-apiserver/app/server.go

```diff
@@ -244,6 +245,12 @@ func CreateNodeDialer(s completedServerRunOptions) (tunneler.Tunneler, *http.Tra
 		// Get ssh key distribution func, if supported
 		var installSSHKey tunneler.InstallSSHKey
 
+		if utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) && cloudprovider.IsDeprecatedInternal(s.CloudProvider.CloudProvider) {
+			return nil, nil, fmt.Errorf(
+				"cloud provider %q was specified, but built-in cloud providers are disabled. Please set --cloud-provider=external and migrate to an external cloud provider",
```

Comment:
Just setting --cloud-provider=external is insufficient for this error case. We are in this code path because the "--ssh-user" (https://github.com/kubernetes/kubernetes/blob/cbe41b4e76ac9f2f70c59c80b02b19e7a935f5cf/cmd/kube-apiserver/app/options/options.go#L200) flag was set. For this use case, we also need to also ask the user to unset it. Without the cloud we won't set the installSSHKey and that means the tunnel will error out later with a less decipherable message.

Reply:
Might be nice to reference https://github.com/kubernetes-sigs/apiserver-network-proxy as an alternate solution.